api/src/utils/validate-theme-json.ts

```diff
@@ -0,0 +1,72 @@
+import Joi from 'joi';
+import { Theme } from '@directus/shared/types';
+
+/**
+ * The following Joi definitions align with the theme
+ * type definition in shared > types > theme.ts
+ */
+
+// Matches hex values i.e. #000000
+const hex = Joi.string()
+	.trim()
+	.pattern(/^#(([\da-fA-F]{3}){1,2}|([\da-fA-F]{4}){1,2})$/, 'Hexadecimal color');
+// Matched css variable references i.e. var(--g-color-background-normal)
+const link = Joi.string()
+	.trim()
+	.pattern(/^var\(\s*--[-\w]+\s*\)$/, 'CSS variable link');
+
+// Arrays of strings. Will be joined by commas on style generation.
+const list = Joi.array().items(Joi.string());
+
+const baseColorVariants = Joi.object({
+	normal: Joi.alternatives([hex, link]),
+	accent: Joi.alternatives([hex, link]),
+	subtle: Joi.alternatives([hex, link]),
+});
+
+const globalSettings = Joi.object({
+	font: Joi.object({
+		family: Joi.object({
+			sans: Joi.alternatives([Joi.string(), list]),
+			serif: Joi.alternatives([Joi.string(), list]),
+			mono: Joi.alternatives([Joi.string(), list]),
+		}),
+	}),
+	color: Joi.object({
+		primary: baseColorVariants,
+		secondary: baseColorVariants,
+		success: baseColorVariants,
+		warning: baseColorVariants,
+		danger: baseColorVariants,
+		foreground: baseColorVariants.append({
+			invert: Joi.alternatives([hex, link]),
+		}),
+		background: baseColorVariants.append({
+			page: Joi.alternatives([hex, link]),
+			invert: Joi.alternatives([hex, link]),
+		}),
+		border: baseColorVariants,
+	}),
+	border: Joi.object({
+		width: Joi.number(),
```

Comment:
`border` and `width` should have a sane min and max value. "accidentally" setting a border width of 2000 will make the app unusable requiring a database edit (or api call) to fix.
```suggestion
		width: Joi.number().min(0).max(32),
```